xdsl/traits.py

```diff
@@ -203,30 +203,51 @@ class SymbolOpInterface(OpTrait):
     a `SymbolTable` (TODO). A Symbol operation should use the SymbolOpInterface interface to
     provide the necessary verification and accessors.
 
-    Currently the only requirement is a "sym_name" attribute of type StringAttr.
+    A Symbol operation may be optional or not. If - the dault - it is not optional,
+    a `sym_name` attribute of type StringAttr is required. If it is optional,
+    the attribute is optional too.
 
-    Please see MLIR documentation for Symbol and SymbolTable for the requirements that are
-    upcoming in xDSL.
+    xDSL offers OptionalSymbolOpInterface as an always-optional SymbolOpInterface helper.
+
+    More requirements are defined in MLIR; Please see MLIR documentation for Symbol and
+    SymbolTable for the requirements that are upcoming in xDSL.
 
     https://mlir.llvm.org/docs/SymbolsAndSymbolTables/#symbol
     """
 
     @staticmethod
-    def get_sym_attr_name(op: Operation) -> StringAttr:
+    def get_sym_attr_name(op: Operation) -> StringAttr | None:
         """
-        Returns the symbol of the operation
+        Returns the symbol of the operation, if any
         """
         # import builtin here to avoid circular import
         from xdsl.dialects.builtin import StringAttr
 
+        concrete = op.get_trait(SymbolOpInterface)
+        assert concrete is not None
+        if concrete.is_optional_symbol(op) and "sym_name" not in op.attributes:
+            return None
         attr = op.attributes["sym_name"]
         assert isinstance(attr, StringAttr)
         return attr
 
+    @staticmethod
+    def is_optional_symbol(op: Operation) -> bool:
```

Comment:
```suggestion
    def is_optional_symbol(self, op: Operation) -> bool:
```
Somehow, I would feel that passing `self` would make more sense. I feel that this would make it less error-prone, by requiring a trait object.

Reply:
Oh, so we avoid people just calling SynbolOpInterface on the op rather than its concrete interface?

Reply:
The reason I prefer calling the `get_trait` on the op directly is that this is less error-prone than just calling it on any operation, and then trigerring an assert/exception in the `is_optional_symbol` function.